apps/api-gateway/src/public-data/public-data.service.spec.ts

```diff
@@ -0,0 +1,75 @@
+import { NotFoundException } from '@nestjs/common';
+import { ClientProxy } from '@nestjs/microservices';
+import { Test, TestingModule } from '@nestjs/testing';
+import { of, firstValueFrom, lastValueFrom, throwError } from 'rxjs';
+import { PublicDataService } from './public-data.service';
+
+describe('PublicDataService', () => {
+  let service: PublicDataService;
+  let clientProxy: ClientProxy;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        PublicDataService,
+        {
+          provide: 'PUBLIC_DATA_SERVICE',
+          useValue: {
+            send: jest.fn(),
+          },
+        },
+      ],
+    }).compile();
+
+    service = module.get<PublicDataService>(PublicDataService);
+    clientProxy = module.get<ClientProxy>('PUBLIC_DATA_SERVICE');
+  });
+
+  it('모듈 정의', () => {
+    expect(service).toBeDefined();
+  });
+
+  it('전체 post_category 데이터 조회', async () => {
+    const mockPostData = [{ id: 1, post_category_name: '팀원 모집' }];
+    jest.spyOn(clientProxy, 'send').mockImplementation(() => of(mockPostData));
+
+    const result = await firstValueFrom(service.getPostData());
+    expect(result).toEqual(mockPostData);
+    expect(clientProxy.send).toHaveBeenCalledWith({ cmd: 'get-post-data' }, {});
+  });
+
+  it('특정 post_category 데이터 조회', async () => {
+    const mockPostData = { id: 1, post_category_name: '팀원 모집' };
+    jest.spyOn(clientProxy, 'send').mockImplementation(() => of(mockPostData));
+
+    const result = await lastValueFrom(service.getOnePostData(1));
+    expect(result).toEqual(mockPostData);
+    expect(clientProxy.send).toHaveBeenCalledWith({ cmd: 'get-one-post-data' }, { id: 1 });
```

Comment:
mock data가 변경된다고 했을 때 관리 포인트가 너무 많은 것 같아서 아래처럼 표현해주면 좋을 것 같아요.
```suggestion
    const mockPostData = { id: 1, post_category_name: '팀원 모집' };
    jest.spyOn(clientProxy, 'send').mockImplementation(() => of(mockPostData));

    const result = await lastValueFrom(service.getOnePostData(mockPostData.id));
    expect(result).toEqual(mockPostData);
    expect(clientProxy.send).toHaveBeenCalledWith({ cmd: 'get-one-post-data' }, { id: mockPostData.id });
```

Reply:
만들어 놓은 변수나 값들을 활용하지 않고 단순히 하드코딩을 많이 활용했네요...
해당 부분도 유기적으로 작성된 변수나 값들을 통해 테스트 하도록
변경하겠습니다!